### PR TITLE
Checkout: Remove productsList factory and use Redux

### PR DESCRIPTION
**Pull request by aidvu**
- Remove the productsList factory from the controller
- Update components to use `QueryComponents` for the `productsList`
- Minor refactoring, assigning unique keys to GApps component, and passing booleans properly
- ES6ify Checkout component

Test:
- Successfully get through checkout, and buy a domain
- Test the GSuite Nudge (not sure anymore if it's still an A/B test, but can hit it directly after checkout `/checkout/:site/with-gsuite/:domain`)
- Try to add a G Suite email through Domains -> Add and see no warnings in console. :)

### CODE REVIEW

**Comment by aidvu** [comment]
Here you go @hambai. :)

File: client/components/upgrades/google-apps/google-apps-dialog/users.jsx [hunk 1]
```diff
@@ -55,11 +55,12 @@ class GoogleAppsUsers extends React.Component {
 		return (
 			<GoogleAppsUsersForm
 				domain={ this.props.domain }
+				index={ index }
+				key={ `google-apps-user-form-${ index }` }
```
Comment (klimeryk) [inline client/components/upgrades/google-apps/google-apps-dialog/users.jsx]: ❤️

**Comment by klimeryk** [comment]
Was testing something else and I think this introduced a regression.
Steps to reproduce:
* Add something to cart, leave for some time (?) come back and see the "Your site deserves a boost!" notice.
* Click the `Complete the purchase` link.
* Error in the console:
```
client.js:62 [HMR] connected
privacy-protection.jsx:75 Uncaught TypeError: Cannot read property 'cost_display' of undefined
    at PrivacyProtection.getPrivacyProtectionCost (privacy-protection.jsx:75)
    at PrivacyProtection.render (privacy-protection.jsx:140)
    at finishClassComponent (react-dom.development.js:7873)
    at updateClassComponent (react-dom.development.js:7850)
    at beginWork (react-dom.development.js:8225)
    at performUnitOfWork (react-dom.development.js:10224)
    at workLoop (react-dom.development.js:10288)
    at HTMLUnknownElement.callCallback (react-dom.development.js:542)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:581)
    at invokeGuardedCallback (react-dom.development.js:438)
```
The problematic line: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/privacy-protection.jsx#L28

**Comment by aidvu** [comment]
Mega rare, but will add a check for it.

**Comment by aidvu** [comment]
Made a fix https://github.com/Automattic/wp-calypso/pull/23445.

**Pull request merged**